grakn-graph/src/main/java/ai/grakn/graph/internal/ConceptImpl.java

```diff
@@ -108,40 +93,75 @@ public void delete() throws GraphOperationException {
      * @param value The new value of the unique property
      * @return The concept itself casted to the correct interface itself
      */
-    T setUniqueProperty(Schema.ConceptProperty key, String value){
-        if(!getGraknGraph().isBatchGraph()) {
-            Concept fetchedConcept = getGraknGraph().getConcept(key, value);
+    T propertyUnique(Schema.ConceptProperty key, String value){
+        if(!vertex().graph().isBatchGraph()) {
+            Concept fetchedConcept = vertex().graph().getConcept(key, value);
             if (fetchedConcept != null) throw PropertyNotUniqueException.cannotChangeProperty(this, fetchedConcept, key, value);
         }
 
-        return setProperty(key, value);
+        return property(key, value);
     }
 
     /**
      * Deletes the node and adds it neighbours for validation
      */
     void deleteNode(){
-        getGraknGraph().getTxCache().remove(this);
-        // delete node
-        getVertex().remove();
+        graph().txCache().remove(this);
+        vertex().delete();
     }
 
     /**
      *
-     * @param edgeType The edge label to traverse
-     * @return The neighbouring concepts found by traversing outgoing edges of a specific type
+     * @param direction the direction of the neigouring concept to get
+     * @param label The edge label to traverse
+     * @return The neighbouring concepts found by traversing edges of a specific type
      */
-    <X extends Concept> Stream<X> getOutgoingNeighbours(Schema.EdgeLabel edgeType){
-        return getEdgesOfType(Direction.OUT, edgeType).map(EdgeElement::getTarget);
+    <X extends Concept> Stream<X> neighbours(Direction direction, Schema.EdgeLabel label){
+        Stream<X> edges;
+
+        switch (direction){
+            case BOTH:
+                edges = vertex().getEdgesOfType(direction, label).
+                        flatMap(edge -> Stream.of(
+                                vertex().graph().factory().buildConcept(edge.source()),
+                                vertex().graph().factory().buildConcept(edge.target())
+                        ));
+                break;
+            case IN:
+                edges = vertex().getEdgesOfType(direction, label).map(edge ->
+                        vertex().graph().factory().buildConcept(edge.source())
+                );
+                break;
+            case OUT:
+                edges = vertex().getEdgesOfType(direction, label).map(edge ->
+                        vertex().graph().factory().buildConcept(edge.target())
+                );
+                break;
+            default:
+                throw GraphOperationException.invalidDirection(direction);
```

Comment:
This isn't in master yet, but in my Graql exception refactor I have `CommonUtil.unreachableStatement()` for these kinds of situations.